From 9c2278dad498b8e4040f30c80cf65b3a089ba218 Mon Sep 17 00:00:00 2001
From: talyz <kim.lindberger@gmail.com>
Date: Fri, 14 Feb 2020 16:26:36 +0100
Subject: [PATCH] Build tests again

The tests were accidentally disabled in
688095d0a7d22704b5c3282bc68b41ceca42ab7e. Since then, the code has
drifted slightly: the synergy lib has been renamed from synergy to
synlib in 4263fd17177d7717b04ac6d6ec62efa2f657ed74 and the curl
dependency was dropped in 491bb2de000245a943b8298462c4a9d8f34c9a44.

This reenables the tests, targets the right lib and removes the
obsolete test.
---
 src/CMakeLists.txt                            |  2 +
 src/test/integtests/CMakeLists.txt            |  2 +-
 .../integtests/arch/ArchInternetTests.cpp     | 37 -------------------
 src/test/unittests/CMakeLists.txt             |  2 +-
 4 files changed, 4 insertions(+), 39 deletions(-)
 delete mode 100644 src/test/integtests/arch/ArchInternetTests.cpp

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index ab63a066..fee080ab 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -22,3 +22,5 @@ add_subdirectory(cmd)
 if (SYNERGY_BUILD_LEGACY_GUI)
 add_subdirectory(gui)
 endif (SYNERGY_BUILD_LEGACY_GUI)
+
+add_subdirectory(test)
diff --git a/src/test/integtests/CMakeLists.txt b/src/test/integtests/CMakeLists.txt
index f39968a3..096ba3d5 100644
--- a/src/test/integtests/CMakeLists.txt
+++ b/src/test/integtests/CMakeLists.txt
@@ -68,4 +68,4 @@ endif()
 
 add_executable(integtests ${sources})
 target_link_libraries(integtests
-    arch base client common io ipc mt net platform server synergy gtest gmock ${libs} ${OPENSSL_LIBS})
+    arch base client common io ipc mt net platform server synlib gtest gmock ${libs} ${OPENSSL_LIBS})
diff --git a/src/test/integtests/arch/ArchInternetTests.cpp b/src/test/integtests/arch/ArchInternetTests.cpp
deleted file mode 100644
index 95823e9f..00000000
--- a/src/test/integtests/arch/ArchInternetTests.cpp
+++ /dev/null
@@ -1,37 +0,0 @@
-/*
- * synergy -- mouse and keyboard sharing utility
- * Copyright (C) 2014-2016 Symless Ltd.
- *
- * This package is free software; you can redistribute it and/or
- * modify it under the terms of the GNU General Public License
- * found in the file LICENSE that should have accompanied this file.
- *
- * This package is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
-
-#include "arch/Arch.h"
-
-#include "test/global/gtest.h"
-
-#define TEST_URL "https://symless.com/tests/?testString"
-//#define TEST_URL "http://localhost/synergy/tests/?testString"
-
-TEST(ArchInternetTests, get)
-{
-    ARCH_INTERNET internet;
-    String result = internet.get(TEST_URL);
-    ASSERT_EQ("Hello world!", result);
-}
-
-TEST(ArchInternetTests, urlEncode)
-{
-    ARCH_INTERNET internet;
-    String result = internet.urlEncode("hello=+&world");
-    ASSERT_EQ("hello%3D%2B%26world", result);
-}
diff --git a/src/test/unittests/CMakeLists.txt b/src/test/unittests/CMakeLists.txt
index 54131eb2..46307e90 100644
--- a/src/test/unittests/CMakeLists.txt
+++ b/src/test/unittests/CMakeLists.txt
@@ -68,4 +68,4 @@ endif()
 
 add_executable(unittests ${sources})
 target_link_libraries(unittests
-    arch base client server common io net platform server synergy mt ipc gtest gmock shared ${libs} ${OPENSSL_LIBS})
+    arch base client server common io net platform server synlib mt ipc gtest gmock shared ${libs} ${OPENSSL_LIBS})
-- 
2.25.0

